rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -341,72 +363,233 @@ def _convert_attribute_tokens_to_texts(
+
+        Returns:
+            Size of additional vocabulary that should be set aside for incremental training.
+        """
+        # Vocabulary expansion for INTENTS, ACTION_NAME
+        # and INTENT_RESPONSE_KEY is currently not supported as
+        # incremental training does not support creation/deletion
+        # of new/existing labels(intents, actions, etc.)
+        if attribute not in DENSE_FEATURIZABLE_ATTRIBUTES:
+            return 0
+        if self.additional_vocabulary_size.get(attribute) is not None:
+            return self.additional_vocabulary_size[attribute]
+
+        # If the user hasn't defined additional vocabulary size,
+        # then we increase it by 1000 minimum. If the current
+        # vocabulary size is greater than 2000, we take half of
+        # that number as additional vocabulary size.
+        return max(MIN_ADDITIONAL_CVF_VOCABULARY, int(existing_vocabulary_size * 0.5))
+
+    def _add_buffer_to_vocabulary(self, attribute: Text) -> None:
+        """Add extra tokens to vocabulary for incremental training.
+
+        These extra tokens act as buffer slots which are used up sequentially
+        when more data is received as part of incremental training. Each of
+        these tokens start with a prefix `buf_` followed by the extra slot index.
+        So for example - buf_1, buf_2, buf_3... and so on.
+
+        Args:
+            attribute: Name of the attribute for which the vocabulary should be expanded.
+
+        """
+        original_vocabulary = self.vectorizers[attribute].vocabulary_
+        current_vocabulary_size = len(original_vocabulary)
+        for index in range(
+            current_vocabulary_size,
+            current_vocabulary_size
+            + self._get_additional_vocabulary_size(attribute, current_vocabulary_size),
+        ):
+            original_vocabulary[f"{BUFFER_SLOTS_PREFIX}{index}"] = index
+        self._set_vocabulary(attribute, original_vocabulary)
+
+    def _set_vocabulary(
+        self, attribute: Text, original_vocabulary: Dict[Text, int]
+    ) -> None:
+        """Set the vocabulary of the vectorizer of attribute.
+
+        Args:
+            attribute: Message attribute for which vocabulary should be set
+            original_vocabulary: Vocabulary for the attribute to be set.
+
+        """
+        self.vectorizers[attribute].vocabulary_ = original_vocabulary
+        self.vectorizers[attribute]._validate_vocabulary()
+
+    @staticmethod
+    def _construct_vocabulary_from_texts(
+        vectorizer: CountVectorizer, texts: List[Text]
+    ) -> Set:
+        """Apply vectorizer's preprocessor on texts to get the vocabulary from texts.
+
+        Args:
+            vectorizer: Sklearn's count vectorizer which has been pre-configured.
+            texts: Examples from which the vocabulary should be constructed
+
+        Returns:
+            Set of unique vocabulary words extracted.
```

Comment:
I'd avoid to repeat the type in the docstring. That just makes it harder to maintain and the `unique` is already specifying the interesting part.
```suggestion
            Unique vocabulary words extracted.
```